apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -129,30 +129,65 @@ private void MaintainPool()
         /// </summary>
         /// <returns>A snapshot of statistics for this pool.</returns>
         public Statistics GetStatisticsSnapshot() =>
-            new Statistics(_targetedPools.ToArray().Select(p => p.Value.GetStatisticsSnapshot()).ToList().AsReadOnly());
-
+            new Statistics(_targetedPools.ToArray().Select(p => p.Value.GetStatisticsSnapshot()).ToList().AsReadOnly());
+
         /// <summary>
         /// Provides a snapshot of statistics for a database-specific pool.
+        /// This is equivalent of calling <see cref="GetPoolStatisticsSnapshot(SessionPoolKey)"/> passing a session pool key
+        /// with a null database role.
         /// </summary>
         /// <returns>A snapshot of statistics for this pool.</returns>
-        public DatabaseStatistics GetStatisticsSnapshot(DatabaseName databaseName)
-        {
-            GaxPreconditions.CheckNotNull(databaseName, nameof(databaseName));
-            return _targetedPools.TryGetValue(databaseName, out var pool) ? pool.GetStatisticsSnapshot() : null;
-        }
+        [Obsolete("Use the overload GetPoolStatisticsSnapshot(DatabaseName) instead.")]
+        public DatabaseStatistics GetStatisticsSnapshot(DatabaseName databaseName)
+        {
+            var poolStatistics = GetPoolStatisticsSnapshot(databaseName);
+            return (poolStatistics is null) ? null : new DatabaseStatistics(poolStatistics);
+        }
+
+        /// <summary>
+        /// Provides a snapshot of statistics for pool associated with the given <see cref="SessionPoolKey"/>.
```

Comment:
Change here and everywhere.

```suggestion
        /// Provides a snapshot of statistics for the pool associated with the given <see cref="SessionPoolKey"/>.
```